### ISSUE TITLE
Quart will not auto-reload on code changes when run via docker container

### ISSUE THREAD

**Issue by larsblumberg** [pallets/quart#191]
Hello! I am using latest `quart==0.18.0` When running `quart` in debug mode via `QUART_DEBUG=true` and `QUART_ENV=development`, `quart` will not auto-reload when running in a docker container. The source directory is volume-mounted properly into the container. A similar issue affecting `hypercorn` also been reported here: https://stackoverflow.com/questions/67506424/hypercorn-with-reload-and-docker-volumes, albeit with no solution yet.

**docker-compose.yaml**
```yaml
services:
  app:
    build: .
    command: quart run --host=0.0.0.0 --port=8080
    environment:
      QUART_APP: src:app
      QUART_DEBUG: true
      QUART_ENV: development
    ports:
      - 8080:8080
    volumes:
      - ./src:/app/src
```

**Dockerfile**

```dockerfile
FROM python:3.10-slim-bullseye
WORKDIR /app                                                          
RUN pip install hypercorn==0.14.3 quart==0.18.0
COPY src ./src
EXPOSE 8080
# This is the production command; docker-compose.yaml overwrites it for local development
CMD hypercorn --bind 0.0.0.0:8080 src:app
```

**src/__init__.py**

```python
from quart import Quart

app=Quart(__name__)

@app.route('/', methods=['GET'])
def get_root():
    return "Hello world!"
```

Now run the stack via `docker-compose up` and change `src/__init__.py`. `quart` will not reload.

If however the `command` in `docker-compose.yaml` is replaced to use `hypercorn --reload`, the server _will_ reload on code changes!

```yaml
services:
  app:
    build: .
    command: hypercorn --bind 0.0.0.0:8080 --reload src:app
    # Rest as above
    # ...
```

**Comment by pgjones** [comment]
This should be fixed by 7634a5a56b757eb6c5602407114d587ace8f5590 (not yet released)

**Comment by pgjones** [comment]
Now released in 0.18.1
